### ISSUE TITLE
"salt '*' test.ping" run on master has terminated salt-minion on a group of minions

### ISSUE THREAD

**Issue by Eishzigh** [saltstack/salt#41424]
### Description of Issue/Question
On 2017-05-25 14:45 on Master "salt '*' test.ping" reported some minions (over 20) as " Minion did not return. [Not connected]".

On all of those minions last record salt-minion appeared not running, and 'tail /var/log/salt/minion' was 
2017-05-25 14:45:03,426 [salt.crypt                                           ][CRITICAL][14810] The Salt Master has rejected this minion's public key!
To repair this issue, delete the public key for this minion on the Salt Master and restart this minion.
Or restart the Salt Master in open mode to clean out the keys. The Salt Minion will now exit.

I fixed the issue with deleting their keys from both master and minions, and restarting service salt-minion.

MASTER: salt-key -d MINION
MINION: rm /etc/salt/pki/minion/* ; service salt-minion restart
MASTER: salt-key -a MINION

### Setup
on the Master configuration defines only
worker_threads, file_roots, and nodegroups

on some rejected Minions configuration defines only hostname of the Master, 
on other rejected Minions the master hostname is not defined (default).
 
### Steps to Reproduce Issue
I cannot reproduce this yet.  After recreating public keys on the minions and pushing them to the master, all the minions keep responding.

### Versions Report

# salt --versions-report
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.4.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: 0.3.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.3
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.9 Santiago
        machine: x86_64
        release: 2.6.32-642.15.1.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.9 Santiago

All rejected minions are running RHEL6.8

# salt-minion --versions-report
Salt Version:
           Salt: 2016.11.4

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.4.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: unknown
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: 0.3.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.3.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 3.2.5

System Versions:
           dist: redhat 6.8 Santiago
        machine: x86_64
        release: 2.6.32-642.13.1.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.8 Santiago

**Comment by gtmanfred** [comment]
Did you update to from salt 2016.11.4 to 2016.11.5 right before this update?

Thanks,
Daniel

**Comment by Eishzigh** [comment]
Yes, we did.

**Comment by gtmanfred** [comment]
I am wondering if this was caused by us switching back to pycrypto on rhel 6 boxes.  For 2016.11.4, we switched to pycryptodomex, but didn't take every case into account.  There were some problems with fips enabled machines.

My guess is that the switch back may have caused the problem?  I am going to leave this open to see if we see the same problem with anyone else, but I have been unable to replicate the error.

Thanks,
Daniel
